services/wallet/transfer/commands_sequential.go

```diff
@@ -43,43 +41,12 @@ func (c *findBlocksCommand) Command() async.Command {
 }
 
 func (c *findBlocksCommand) Run(parent context.Context) (err error) {
-	log.Info("start findBlocksCommand", "account", c.account, "chain", c.chainClient.ChainID, "noLimit", c.noLimit)
+	log.Debug("start findBlocksCommand", "account", c.account, "chain", c.chainClient.ChainID, "noLimit", c.noLimit)
 
 	rangeSize := big.NewInt(DefaultNodeBlockChunkSize)
 
-	to, err := c.loadFirstKnownBlockNumber()
-	log.Info("findBlocksCommand", "firstKnownBlockNumber", to, "error", err)
-
-	if err != nil {
-		if err.Error() != allBlocksLoaded {
-			c.error = err
-		}
-
-		return nil // We break the loop if we fetched all the blocks
-	}
-
-	var head *types.Header = nil
-
-	if to == nil {
-		ctx, cancel := context.WithTimeout(parent, 3*time.Second)
-		head, err = c.chainClient.HeaderByNumber(ctx, nil)
-		cancel()
-
-		if err != nil {
-			c.error = err
-			log.Error("findBlocksCommand failed to get head block", "error", err)
-			return nil
-		}
-
-		log.Info("current head is", "chain", c.chainClient.ChainID, "block number", head.Number)
-
-		to = new(big.Int).Set(head.Number) // deep copy
-	} else {
-		to.Sub(to, big.NewInt(1))
-	}
-
-	var from = big.NewInt(0)
-	if to.Cmp(rangeSize) > 0 {
+	from, to := new(big.Int).Set(c.fromBlockNumber), new(big.Int).Set(c.toBlockNumber)
+	if to.Cmp(from) > DefaultNodeBlockChunkSize {
```

Comment:
```
// Cmp compares x and y and returns:
//
//	-1 if x <  y
//	 0 if x == y
//	+1 if x >  y
```

Don't you want to do a `Sub` between `from` and `to`, and then a `Cmp` with `DefaultNodeBlockChunkSize`?